bin/combinators.ml

```diff
@@ -1,14 +1,59 @@
-Random.init 2025
+(* random int generators are from qcheck 
+  - allows test results to be consistent
+  - shortcut for the function signature 
+  I'll probably create my own version tho *)
 
-let rec int_list s = 
+let rec int_list_size_gen s st = 
   if s <= 0 then
     []
   else
-    Random.int 100000 :: int_list (s - 1)
+    QCheck.Gen.int st :: int_list_size_gen (s - 1) st
 
-let rec print_list l = match l with 
-| [] -> print_newline ()
-| h ::t -> print_int h; print_char ' '; print_list t
+(* int list sorted in ascending order *)
+let rec int_list_sorted_gen prev s st = 
+  if s <= 0 then
+    []
+  else
+    let n = QCheck.Gen.int st in
+    if n >= prev then
+      n :: int_list_sorted_gen n (s - 1)st
+    else 
+      int_list_sorted_gen prev (s - 1) st
+
+(* int list with each element identical *)
+let int_list_dup_gen s st = 
+  let n = QCheck.Gen.int st in
+  let rec aux s' st' =
+    if s' <= 0 then
+      []
+    else
+        n :: aux (s' - 1) st' in
+  aux s st
+
+(* int list with each element unique *)
+(* either dup or unique needs more work *)
+let int_list_unique_gen s st = 
+  let set = Hashtbl.create s in
+  let rec aux s' st' =
+    if s' <= 0 then
+      []
+    else
+      let n = QCheck.Gen.int st' in
+      (fun x -> Hashtbl.replace set x ()) n;
```

Comment:
You don't need to create a lambda and then immediately apply it. A simple `Hashtbl.replace set n ();` is fine.